appium/appium
# Pipeline to release Appium from Azure

## Proposed changes

This will make it so you can release Appium (rc or latest) from https://dev.azure.com/AppiumCI/Appium%20CI/_build/results?buildId=3965

What it does
* Sets the user to github.com/triager
* Checks out $RELEASE_BRANCH, a parameter that the user sets. (This would normally be `1.14`, `1.15`, etc...)
* Sets the version to $APPIUM_VERSION in package.json and npm-shrinkwrap.json, and then commits and pushes to $RELEASE_BRANCH
* Creates the Git tag $APPIUM_VERSION
* Publishes $APPIUM_VERSION. Parses the preId as the "channel". Example `1.15.0-rc.5` will run `npm publish --tag rc`; `1.16.0` will run `npm publish`

To test it out yourself:

* Go to https://dev.azure.com/AppiumCI/Appium%20CI/_build?definitionId=40 (requires sign-in with access)
* Click 'Queue'
* Select the `APPIUM_VERSION` as `<major>.<minor>.<patch>-fake.<unique-number>` (e.g.: `1.15.0-fake.358`)
* Select the `RELEASE_BRANCH` as any branch (note that this pipeline makes a commit to that branch)

## Types of changes

What types of changes does your code introduce to Appium?
_Put an `x` in the boxes that apply_


## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [CONTRIBUTING](https://github.com/appium/appium/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla.js.foundation/appium/appium)
